packages/block-editor/src/components/provider/use-block-sync.js

```diff
@@ -96,8 +98,18 @@ export default function useBlockSync( {
 		if ( clientId ) {
 			setHasControlledInnerBlocks( clientId, true );
 			__unstableMarkNextChangeAsNotPersistent();
-			replaceInnerBlocks( clientId, controlledBlocks );
+			// This needs to be called after the setHasControlledInnerBlocks call
```

Comment:
Should it also say why?

Reply:
Good catch, this is actually not needed anymore, it's an artifact of one of my earlier explorations.